lib/calendar_recurrence.ex

```diff
@@ -94,17 +99,31 @@ end
 defprotocol CalendarRecurrence.T do
   def continue?(t1, t2)
 
-  def add(t, count)
+  def add(t, count, unit)
 
-  def diff(t1, t2)
+  def diff(t1, t2, unit)
 end
 
 defimpl CalendarRecurrence.T, for: Date do
   def continue?(date1, date2) do
     Date.compare(date1, date2) in [:lt, :eq]
   end
 
+  def add(date, step, _unit), do: add(date, step)
```

Comment:
could you change this to:

```suggestion
  def add(date, step, _unit), do: Date.add(date, step)
```

and then remove the redundant `defdelegate add(date, step)` below? Ditto below.

Reply:
Sure!